presto/presto.go

```diff
@@ -613,7 +627,9 @@ var _ driver.Rows = &driverRows{}
 
 func (qr *driverRows) Close() error {
 	if qr.nextURI != "" {
-		req, err := qr.conn.newRequest("DELETE", qr.nextURI, nil, nil)
+		hs := make(http.Header)
+		hs.Add(prestoUserHeader, qr.user)
```

Comment:
I don't think we want to add the user header here, it's probably better off in the newRequest function only when kerberos is being used

Reply:
each time calling newRequest, it should set X-PRESTO-USERS, without the code change, it always use the one parsing from PrestoURI of config, which is ONLY initialized when creating the driver. 

While the single driver should be able to handle different statements by a different user, which means, people who use this library should be able to authenticate the end user by some way and use the same connection to issue a query for the different user.

Then the single query needs a POST, following by several GET and finally a DELETE, they both need to use the (original end user) to do this. That's the reason to implement this.

Do you have other suggestions?

Reply:
So, frankly, I suspected some sort of one-of-a-kind snowflaky use case for this, but accordingly to @BlueStalker's explanation below, this actually looks pretty reasonable.